api/system/v1alpha1/zone_insight_helpers.go

```diff
@@ -74,6 +74,17 @@ func (x *ZoneInsight) UpdateSubscription(s generic.Subscription) error {
 	return nil
 }
 
+// CompactFinished removes detailed information about finished subscriptions to trim the object size
+// The last subscription always has details.
+func (x *ZoneInsight) CompactFinished() {
+	for i := 0; i < len(x.GetSubscriptions())-1; i++ {
```

Comment:
Should we check if `DisconnectTime` is not `nil`? I think it's possible to have a period of time when there are 2 open subscriptions, and maybe we don't want to cleanup too early

Reply:
technically only the last should be not finished. If this is not the case, we have a bug in different place